src/vs/editor/contrib/unusualLineTerminators/browser/unusualLineTerminators.ts

```diff
@@ -85,13 +86,25 @@ export class UnusualLineTerminatorsDetector extends Disposable implements IEdito
 			return;
 		}
 
-		const result = await this._dialogService.confirm({
-			title: nls.localize('unusualLineTerminators.title', "Unusual Line Terminators"),
-			message: nls.localize('unusualLineTerminators.message', "Detected unusual line terminators"),
-			detail: nls.localize('unusualLineTerminators.detail', "The file '{0}' contains one or more unusual line terminator characters, like Line Separator (LS) or Paragraph Separator (PS).\n\nIt is recommended to remove them from the file. This can be configured via `editor.unusualLineTerminators`.", basename(model.uri)),
-			primaryButton: nls.localize('unusualLineTerminators.fix', "Remove Unusual Line Terminators"),
-			secondaryButton: nls.localize('unusualLineTerminators.ignore', "Ignore")
-		});
+		if (this._isPresentingDialog) {
+			// we're currently showing the dialog, which is async.
+			// avoid spamming the user
+			return;
+		}
+
+		let result: IConfirmationResult;
+		try {
+			this._isPresentingDialog = true;
+			result = await this._dialogService.confirm({
+				title: nls.localize('unusualLineTerminators.title', "Unusual Line Terminators"),
+				message: nls.localize('unusualLineTerminators.message', "Detected unusual line terminators"),
+				detail: nls.localize('unusualLineTerminators.detail', "The file '{0}' contains one or more unusual line terminator characters, like Line Separator (LS) or Paragraph Separator (PS).\n\nIt is recommended to remove them from the file. This can be configured via `editor.unusualLineTerminators`.", basename(model.uri)),
+				primaryButton: nls.localize('unusualLineTerminators.fix', "Remove Unusual Line Terminators"),
+				secondaryButton: nls.localize('unusualLineTerminators.ignore', "Ignore")
+			});
+		} finally {
+			this._isPresentingDialog = false;
+		}
 
 		if (!result.confirmed) {
```

Comment:
I'm surprised to see TypeScript not complain about this. If the dialog service does throw, wouldn't that make `result` be `undefined`?

Reply:
@joaomoreno There's no `catch` block. If it throws, execution goes out of the method, but before doing that it sets `this._isPresentingDialog` to false.

Reply:
Oh my.